ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh

```diff
@@ -327,14 +327,16 @@ ${required_permissions}
 """
 fi
 
+custom_role_name_json="{}"
 if [[ -n "${AZURE_PERMISSION_FOR_CLUSTER_SP}" ]]; then
     sp_role="${AZURE_PERMISSION_FOR_CLUSTER_SP}"
 else
     create_role_definition_json "${CUSTOM_ROLE_NAME}" "${required_permissions}" "${ROLE_DEFINITION}"
     echo "Creating custom role..."
     create_custom_role "${ROLE_DEFINITION}" "${CUSTOM_ROLE_NAME}"
     # for destroy
-    echo "${CUSTOM_ROLE_NAME}" > "${SHARED_DIR}/azure_custom_role_name"
+    custom_role_name_json=$(echo "${custom_role_name_json}" | jq -c -S ". +={\"cluster\":\"${CUSTOM_ROLE_NAME}\"}")
+    echo "${custom_role_name_json}" >> "${SHARED_DIR}/azure_custom_role_name"
```

Comment:
We are always rewriting `custom_role_name_json`, so here I think we do not need to use `>>`

Reply:
right, I meant to use `>`, updated.